raster-foundry/raster-foundry
# Allow backend handling of planet imagery imports

## Overview

This PR makes it possible to import planet imagery from a list of planet image IDs. It's main changes are:

- include planet type in upload create endpoint and the upload datamodel
- include migration 79 to add to the `upload_type` enum in postgres
- add PLANET to the swagger enum for `upload_type`
- add planet scenes generator

### Checklist

- ~Styleguide updated, if necessary~

## Testing Instructions

 * Run migrations, then bring up your server and airflow
 * Use the planet labs explorer to find some imagery you like - https://www.planet.com/explorer/
 * Find the IDs by clicking `API {:}` at the bottom of the imagery list:

![image](https://user-images.githubusercontent.com/5702984/28943973-c6fc8348-786e-11e7-91ce-b2560a720345.png)

 * Create an upload with those image IDs in the `files` field, and filling in the planet API key. Post to `/api/uploads/`

```json
{
    "organizationId":"dfac6307-b5ef-43f7-beda-b9f208bb7726",
    "uploadStatus":"uploaded",
    "fileType":"geotiff",
    "datasource":"c14c8e97-ba85-4677-ac9c-069cfef1f0b1",
    "visibility":"private",
    "uploadType":"planet",
    "files": ["ids", "ids2"],
    "metadata": {"planetKey":"<PLANET API KEY>"},
    "owner": "your user id"
}
```

 * The airflow `process_upload` job should succeed, and you should find imagery in s3 under `user-uploads/<user id>/<upload id>/*.tif`
 * Then ingest should succeed

Closes #2321
